lms/djangoapps/mobile_api/users/tests.py

```diff
@@ -501,21 +501,17 @@ def test_success_v1(self):
         assert response.data['last_visited_block_id'] == str(self.unit.location)
 
     # Since we are testing an non atomic view in atomic test case, therefore we are expecting error on failures
-    def api_error_response(self, reverse_args=None, data=None, **kwargs):
+    def api_atomic_response(self, reverse_args=None, data=None, **kwargs):
         """
-            Same as api response from MobileAPITestCase but handle views which throw errors
+        Same as the api_response from MobileAPITestCase, but handles the view as an atomic transaction.
         """
         url = self.reverse_url(reverse_args, **kwargs)
-        try:
-            with transaction.atomic():
-                self.url_method(url, data=data, **kwargs)
-                assert False
```

Comment:
This test does not seem to be related, but it stopped throwing the `TransactionManagementError` once we added the `site_configuration.get_value()` call to the language preferences middleware.

This was not the case before because this feature was gated behind `DarkLangConfig`, which is disabled in this test. This gating does not make much sense, as this feature does not depend on  `DarkLang`.

Reply:
Nice find @Agrendalath!